internal/stagec1.go

```diff
@@ -0,0 +1,103 @@
+package internal
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/codecrafters-io/kafka-tester/internal/kafka_executable"
+	"github.com/codecrafters-io/kafka-tester/protocol"
+	kafkaapi "github.com/codecrafters-io/kafka-tester/protocol/api"
+	"github.com/codecrafters-io/tester-utils/random"
+	"github.com/codecrafters-io/tester-utils/test_case_harness"
+)
+
+func testSequentialRequests(stageHarness *test_case_harness.TestCaseHarness) error {
+	b := kafka_executable.NewKafkaExecutable(stageHarness)
+	if err := b.Run(); err != nil {
+		return err
+	}
+
+	logger := stageHarness.Logger
+
+	broker := protocol.NewBroker("localhost:9092")
+	if err := broker.ConnectWithRetries(b, logger); err != nil {
+		return err
+	}
+	defer broker.Close()
+
+	requestCount := random.RandomInt(2, 5)
+	for i := 0; i < requestCount; i++ {
+		correlationId := int32(random.RandomInt(-math.MaxInt32, math.MaxInt32))
+		request := kafkaapi.ApiVersionsRequest{
+			Header: kafkaapi.RequestHeader{
+				ApiKey:        18,
+				ApiVersion:    3,
+				CorrelationId: correlationId,
+				ClientId:      "kafka-cli",
+			},
+			Body: kafkaapi.ApiVersionsRequestBody{
+				Version:               3,
+				ClientSoftwareName:    "kafka-cli",
+				ClientSoftwareVersion: "0.1",
+			},
+		}
+
+		message := kafkaapi.EncodeApiVersionsRequest(&request)
+		logger.Infof("Sending \"ApiVersions\" (version: %v) request (Correlation id: %v)", request.Header.ApiVersion, request.Header.CorrelationId)
+
+		response, err := broker.SendAndReceive(message)
+		if err != nil {
+			return err
+		}
+		logger.Infof("Hexdump of sent \"ApiVersions\" request: \n%v\n", protocol.GetFormattedHexdump(message))
```

Comment:
@ryan-gang this applies to the other stages too - hexdumps can be `debug` by default, not `info`. The `Sending ...` log above is info, that should be good enough.

Reply:
This'll get extremely noisy otherwise